### DISCUSSION TITLE
Help with AmazonSES setup

### DISCUSSION THREAD

**Discussion by evrpress** [plausible/analytics#3500] [Self-Hosted Support]
I'm not able to get emails from my self hosted installation (Is there a way to test emails?

I try to reset password to test the email functionality but no email arrives although the account exists:

<img width="397" alt="image" src="https://github.com/plausible/analytics/assets/881729/d62b1e62-d3e1-4183-a18a-1b4bc2d9c6fc">

Here's part of my settings: (Settings are correct, they work somewhere else)

```
MAILER_EMAIL=hello@xxx.xxx
SMTP_HOST_ADDR=email-smtp.us-east-1.amazonaws.com
SMTP_HOST_PORT=587
SMTP_USER_NAME=***
SMTP_USER_PWD=***
SMTP_HOST_SSL_ENABLED=false
```

running `docker logs --tail 100 874fbcd05b84` on the SMTP container returns that

```
    1 LOG: MAIN
    1   exim 4.89 daemon started: pid=1, -q30m, listening for SMTP on port 25 (IPv6 and IPv4)
```

Where should I start?

**Comment by ruslandoga** [comment]
👋 @evrpress 

The `mail` container isn't needed since Amazon SES would be acting as the relay / smart host. You just need to set [the relevant env vars](https://plausible.io/docs/self-hosting-configuration#mailersmtp-setup) for `plausible` container and it should work.

**Reply by evrpress** [reply]
Thanks for the speedy response!

I've posted the SMTP details above. Is this everything which is needed?

**Reply by ruslandoga** [reply]
Are they part of `plausible-conf.env`? If so, they look correct :) Note that if you initially started the container without them, you'd need to recreate the container for the new env vars to take effect.

**Reply by ruslandoga** [reply]
To send an email manually, you can try doing it from within the app

```console
$ cd hosting
$ docker compose exec plausible bin/plausible remote
```
```elixir
iex> # let's first check the config
iex> Application.get_env(:plausible, Plausible.Mailer)
iex> # note you'd need to replace from and to with actual addresses
iex> email = Bamboo.Email.new_email(from: "...", to: "...", subject: "test", text_body: "test")
iex> Plausible.Mailer.deliver_now(email)
```

**Reply by evrpress** [reply]
oh, yes! This is the command to redo the containers

```docker-compose down --remove-orphans && docker-compose pull plausible && docker-compose up -d```

Thanks a lot this helped!

**Reply by mister-px** [reply]
I'm having a similar problem, no mail gets delivered through Amazon SES: I'm not using the "mail" service in docker-compose.yml,
and I correctly configured the env variables, as i can read them correctly in the `Application.get_env(:plausible, Plausible.Mailer)` test line.
After sending the test email with `Plausible.Mailer.deliver_now(email)`, i receive this:
```
15:59:49.064 [warning] Description: 'Server authenticity is not verified since certificate path validation is not enabled'
     Reason: 'The option {verify, verify_peer} and one of the options \'cacertfile\' or \'cacerts\' are required to enable this.'

{:error,
 {:permanent_failure, 'email-smtp.eu-central-1.amazonaws.com',
  "530 Authentication required\r\n"}}
```

I understand the warning is not a problem, but the 530 is. Also, in the AWS console the credentials show up as "Never used", so i guess they are not sent while opening the connection with the SMTP.

Any advice?

**Reply by mister-px** [reply]
Moved to a dedicated discussion (https://github.com/plausible/analytics/discussions/3757)
